### PR TITLE
flow(app): flow type the app reducer

### PR DESCRIPTION
**Pull request by rgbkrk**
Type the `app` reducer this time.

### CODE REVIEW

**Review by jdetle — APPROVED** [review]
Are we eventually going to have no constants for our redux actions?

**Comment by rgbkrk** [comment]
> Are we eventually going to have no constants for our redux actions?

Yup, `flow` determines these for us. We could still have constants if we want them as is though, because `flow` figures out `constants.BLAH` properly too.

**Comment by rgbkrk** [comment]
Couple things this PR made me realize:

* The ZMQ/native local kernel portion of the state should be separate to make exporting easier (can't and won't be used by a web app)
* The github state should be in a separate module too

File: src/notebook/kernel/shutdown.js [hunk 1]
```diff
@@ -1,11 +1,28 @@
...
+  control: Subject,
+}
+
+export type Kernel = {
+  channels: Channels,
+  spawn: any, // ChildProcess,
+  connectionFile: string,
+}
+
+export function cleanupKernel(kernel: Kernel, closeChannels: boolean) {
```
Comment (peggyrayzis) [inline src/notebook/kernel/shutdown.js]: Should we type the returns of these functions? 🤔 Does it even matter?

Comment (rgbkrk) [reply src/notebook/kernel/shutdown.js]: These are void returns with necessary side effects. 😉  I can add some on.

Comment (rgbkrk) [reply src/notebook/kernel/shutdown.js]: Added on some return types. There was one with a `Promise<undefined>` which I switched to `Promise<boolean>` (it was only there to signal completion).

Comment (peggyrayzis) [reply src/notebook/kernel/shutdown.js]: Awesome, this is helpful for understanding what's going on!

**Comment by peggyrayzis** [comment]
Thanks again for another great Flow PR 🎉 !

**Pull request merged**